pkg/tfbridge/defaults/apply.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package defaults
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/hashicorp/go-multierror"
+
+	b "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
+	shim "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+)
+
+// Add mapped resources and datasources according to the given strategies.
+func ApplyStrategy(info *b.ProviderInfo, opts Strategy) error {
```

Comment:
I would splurge on the doc string here as well.


```
Assigns resource and data source tokens unless already specified by the user. 

Context: bridging a provider requires every TF resource and data source have a matching entry in [ProviderInfo.Resources] and [ProviderInfo.DataSources]. Each entry needs to minimally specifying the Pulumi token in [ResourceInfo.Tok]. The mapping entries can be done by hand for smaller provides but become a chore for providers with 1000s of entries. 

ApplyStrategy scans TF resources and datasources to populate missing entries as needed with automatically computed Pulumi tokens. 

ApplyStrategy always respects and does not modify pre-existing entires. The user can therefore manually override the token decision by providing ResourceInfo or DataSourceInfo entry prior to calling ApplyStrategy. 

<TBD discuss on how to prevent resources from being mapped>
```

Reply:
ProviderInfo.IgnoreMappings is respected. If your token is there, it won't be mapped. I added that to the doc comment.